pyracf/resource/resource_admin.py

```diff
@@ -265,11 +265,316 @@ def set_universal_access(
         )
         return self._to_steps(result)
 
+    # ============================================================================
+    # Individual Access
+    # ============================================================================
     def get_my_access(self, resource: str, class_name: str) -> Union[str, bytes, None]:
         """Get the access associated with your own general resource profile."""
         profile = self.extract(resource, class_name, profile_only=True)
         return self._get_field(profile, "base", "yourAccess")
 
+    # ============================================================================
+    # Auditing Rules
+    # ============================================================================
+    def get_audit_rules(
+        self, resource: str, class_name: str
+    ) -> Union[dict, bytes, None]:
+        """Get the auditing rules associated with this general resource profile."""
+        profile = self.extract(resource, class_name, profile_only=True)
+        return self._get_field(profile, "base", "auditing")
+
+    def clear_audit_failures(
+        self, resource: str, class_name: str
+    ) -> Union[dict, bytes]:
+        """
+        Clears the auditing rules for failures for this general resource profile while
+        preserving other rules.
+        """
+        result = [self.extract(resource, class_name)]
+        profile = result[0]["securityResult"]["resource"]["commands"][0]["profiles"][0]
+        audit_rules = self._get_field(profile, "base", "auditing")
+        if "failures" not in audit_rules:
+            return self._to_steps(result[0])
+        traits = {}
+        if "success" in audit_rules:
+            traits[f"base:audit_{audit_rules['success']}"] = "SUCCESS"
+        if "all" in audit_rules:
+            traits[f"base:audit_{audit_rules['all']}"] = "ALL"
+        result.append(self.alter(resource, class_name, traits=traits))
+        return self._to_steps(result)
+
+    def clear_audit_successes(
+        self, resource: str, class_name: str
+    ) -> Union[dict, bytes]:
+        """
+        Clears the auditing rules for successes for this general resource profile while
+        preserving other rules.
+        """
+        result = [self.extract(resource, class_name)]
+        profile = result[0]["securityResult"]["resource"]["commands"][0]["profiles"][0]
+        audit_rules = self._get_field(profile, "base", "auditing")
+        if "success" not in audit_rules:
+            return self._to_steps(result[0])
+        traits = {}
+        if "failures" in audit_rules:
+            traits[f"base:audit_{audit_rules['failures']}"] = "FAILURE"
+        if "all" in audit_rules:
+            traits[f"base:audit_{audit_rules['all']}"] = "ALL"
+        result.append(self.alter(resource, class_name, traits=traits))
+        return self._to_steps(result)
+
+    def clear_audit_both_successes_and_failures(
```

Comment:
I think you can remove `both`

```python
def clear_audit_successes_and_failures(
```

Reply:
I think this is covered by our discussion of the functionality and re-design of these functions. Let me know if this is still a concern.

Reply:
Correct. We no longer have a clear function because the the setters can be used to do this.